File: codigos python/regra_de_tres.py
def regra_de_tres(a, b, c):
    # se o valor de a for 0
    if a == 0:
        return "O valor de 'a' não pode ser zero."
    # calcule o valor de d
    d = (b * c) / a
    return d
# peça ao usuário a , b , c
a = float(input("Digite o valor de a: "))
b = float(input("Digite o valor de b: "))
c = float(input("Digite o valor de c: "))
# mostre o resutado
resultado = regra_de_tres(a, b, c)
print("o valor de d é:",resultado)

File: codigos c/regra_de_tres.c
#include <stdio.h>

float regra_de_tres(float a, float b, float c) {
    if (a == 0) {
        printf("O valor de 'a' não pode ser zero.\n");
        return 0;
    }
    return (b * c) / a;
}

int main() {
    float a, b, c, d;

    printf("Digite o valor de a: ");
    scanf("%f", &a);
    printf("Digite o valor de b: ");
    scanf("%f", &b);
    printf("Digite o valor de c: ");
    scanf("%f", &c);

    d = regra_de_tres(a, b, c);

    if (a != 0) {
        printf("O valor de d é: %.2f\n", d);
    }

    return 0;
}
